### DISCUSSION TITLE
Correct way to assign a role to a group

### DISCUSSION THREAD

**Discussion by arnefromsa** [aws-solutions-library-samples/guidance-for-aws-sustainability-insights-framework#22] [Q&A]
Hi 

I want to update the group permissions of a user after it was created.

Let's say the user looks like the following :

_{
  "email": "user@server.com",
  "state": "active",
  "groups": {
    "/australia": "reader"
  },
  "defaultGroup": "/",
  "createdAt": "2023-08-09T06:09:12.401Z",
  "createdBy": "otheruser@server.com"
}_

Then, I want to perform the update user to be like : 

_{
  "email": "user@server.com",
  "state": "active",
  "groups": {
    "/australia": "reader",
    "/australia/company1": "admin"
  },
  "defaultGroup": "/",
  "createdAt": "2023-08-09T06:09:12.401Z",
  "createdBy": "otheruser@server.com"
}_

When I perform this with the /users/:email PATCH call, and go look at the Cognito groups, it does not reflect the changes in there. What is the correct way to update users' roles on groups ?

**Comment by williamsia** [comment]
Hi @arnefromsa, the `PATCH` endpoint only allows you to modify user's `state` and `password`, to assign user to a group, perform a post operation on a user in a specified group context , so in you case if you want to assign admin to `user@server.com` in group `/australia/company1` then 

perform a POST with this body ( when `x-groupcontextid` header is set to `/australia/company1` group context )

```json
 {
  "email": "user@server.com",
  "role": "admin"
}
```
